pandas-stubs/core/frame.pyi

```diff
@@ -1684,6 +1686,8 @@ class DataFrame(NDFrame, OpsMixin, _GetItemHack):
     @property
     def iloc(self) -> _iLocIndexerFrame[Self]: ...
     @property
+    # mypy complains if we use Index[Any] instead of UnknownIndex here, even though
+    # the latter is aliased to the former ¯\_(ツ)_/¯.
     def index(self) -> UnknownIndex: ...
```

Comment:
this is a complete mystery to me - any ideas?

Reply:
What if you just used `Index` ?

Reply:
Not sure I follow, sorry
- in `timestamps.pyi` it's already using `Index` in this PR (not `UnknownIndex`)
- if I leave it as `Index` in `frame.pyi`, then mypy complains (though pyright is fine with it)

Reply:
> is this what you meant? Because then, the problem persists

yes, that's what I meant.  I guess we have to keep `UnknownIndex` around.

So is the current version ready for final review?

Reply:
yup, thanks!